vignettes/plotting.Rmd

```diff
@@ -0,0 +1,128 @@
+
+```{r CA plotting,  fig.width=8, fig.height=6}
+autoplot(ds$ImportantHiring)
+autoplot(ds$ImportantHiring, "grid")
+autoplot(ds$ImportantHiring, "bar")
+```
+
+Since the autoplot methods produce `ggplot` objects, you can customize and extend them just as you would any other ggplot. For example, you can change the theme of the plot or alter the color palette.
+
+```{r themes, fig.height=6, fig.width=8, warning=FALSE}
+p <- autoplot(ds$CompanySize)
+p + theme_grey()
+p +
+    geom_point(color = "red", size = 3) +
+    scale_x_log10()
+autoplot(ds$ImportantHiring) +
+    scale_color_brewer()
+```
+
+
+You can also define your own themes or use themes from other packages. This is helpful if you would like to send plots to a client using their style guide. For examples of sample corporate themes, take a look at the [ggtech](https://github.com/ricardo-bion/ggtech) package from AirBnb, or the [ggthemes](https://cran.r-project.org/web/packages/ggthemes/vignettes/ggthemes.html) package.
+
+## Extending ggplots
+
+`ggplot2` has a rich ecosystem of extensions which you can use to add functionality to Crunch plots. For example you can use [patchwork](https://github.com/thomasp85/patchwork) to combine several plots together, or add interactivity to your plots with [plotly](https://plot.ly/r/getting-started/). For a full list of what you can do with `ggplot` objects, see the [ggplot extensions](https://www.ggplot2-exts.org/) website.
+
+# Working with CrunchCubes
+
+Crunch datasets are often large enough that you can't easily calculate summary statistics on your local machine. As a result we try to keep as much computation as possible on the server and only send the results of the calculation to your local R session. One of the main ways this is accomplished it through CrunchCubes, which are typically used to calculate cross-tabs between categorical variables.
+
+CrunchCubes are n-dimensional arrays where each dimension corresponds to one of the variables used to calculate the cross tab. For instance `crtabs(~ Professional + CompanySize + Country, ds)` will generate a three dimensional array where each entry is the number of respondents who selected a given category along the three dimensions. You can plot high dimensional CrunchCubes in the same way that you can plot Crunch Variables and the additional dimensions will be assigned to plot facets.
+
+```{r fig.width= 10,  fig.width=8, fig.height=6}
+cube <- crtabs(~ CompanySize + Professional  + TabsSpaces, ds)
+autoplot(cube)
+autoplot(cube, "bar")
+```
+
+Autoplots are great, but cubes can quickly become too complex for a general plotting method
+to do a good job at representing the plot. If we include lots of dimensions in a cube, or include a dimension with a lot of categories the plot becomes unreadable
```

Comment:
I'm not sure it helps us to include an unreadably large plot. Doesn't really help the story, and it means a big API response bundled in the package.